typing_inspect/typing_inspect.py

```diff
@@ -0,0 +1,198 @@
+    """
+
+    return type(tp) is type(ClassVar)
+
+
+def get_origin(tp):
+    """Get the unsubscripted version of a type. Supports generic types, Union,
+    Callable, and Tuple. Returns None for unsupported types. Examples::
+
+        get_origin(int) == None
+        get_origin(ClassVar[int]) == None
+        get_origin(Generic) == Generic
+        get_origin(Generic[T]) == Generic
+        get_origin(Union[T, int]) == Union
+        get_origin(List[Tuple[T, T]][int]) == List
+    """
+
+    if isinstance(tp, type(Generic)):
+        return _gorg(tp)
+    if is_union_type(tp):
+        return Union
+
+    return None
+
+
+def get_parameters(tp):
+    """Return type parameters of a parameterizable type as a tuple
+    in lexicographic order. Parameterizable types are generic types,
+    unions, tuple types and callable types. Examples::
+
+        get_parameters(int) == ()
+        get_parameters(Generic) == ()
+        get_parameters(Union) == ()
+        get_parameters(List[int]) == ()
+
+        get_parameters(Generic[T]) == (T,)
+        get_parameters(Tuple[List[T], List[S]]) == (T, S)
+        get_parameters(Union[S, Tuple[T, T]][int, U]) == (U,)
+        get_parameters(Mapping[T, Tuple[S, T]]) == (T, S)
+    """
+
+    if (
+        is_generic_type(tp) or is_union_type(tp) or
+        is_callable_type(tp) or is_tuple_type(tp)
+    ):
+        return tp.__parameters__
+    return ()
+
+
+def get_last_args(tp):
+    """Get last arguments of (multiply) subscripted type. Examples::
+
+        get_last_args(int) == ()
+        get_last_args(Union) == ()
+        get_last_args(ClassVar[int]) == (int,)
+        get_last_args(Union[T, int]) == (int,)
+        get_last_args(Iterable[Tuple[T, S]][int, T]) == (int, T)
+    """
+
+    if is_classvar(tp):
+        return (tp.__type__,)
+    if is_generic_type(tp) or is_union_type(tp):
+        return tp.__args__
+    return ()
+
+
+def _eval_args(args):
+    """Internal helper for get_args."""
+    res = []
+    for arg in args:
+        if not isinstance(arg, tuple):
+            res.append(arg)
+        elif is_callable_type(arg[0]):
+            if len(arg) == 2:
+                res.append(Callable[[], arg[1]])
+            elif arg[1] is Ellipsis:
+                res.append(Callable[..., arg[2]])
+            else:
+                res.append(Callable[list(arg[1:-1]), arg[-1]])
+        else:
+            res.append(type(arg[0]).__getitem__(arg[0], _eval_args(arg[1:])))
+    return tuple(res)
+
+
+def get_args(tp, evaluate=False):
+    """Get type arguments with all substitutions performed. For unions,
+    basic simplifications used by Union constructor are performed.
+    If `evaluate` is False (default), report result as nested tuple, this matches
+    the internal representation of types. If `evaluate` is True, then all
+    type parameters are applied (this could be expensive). Examples::
+
+        get_args(Union[int, Tuple[T, int]][str]) == (int, (Tuple, str, int))
```

Comment:
There is an example with ``evaluate=True`` 4 lines below, or do you mean _one more_ example?